java/vortex-jni/src/main/java/dev/vortex/api/expressions/proto/Scalars.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.api.expressions.proto;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.NullValue;
+import dev.vortex.proto.ScalarProtos;
+
+final class Scalars {
+    private Scalars() {}
+
+    static ScalarProtos.Scalar nullNull() {
+        return ScalarProtos.Scalar.newBuilder()
+                .setValue(ScalarProtos.ScalarValue.newBuilder()
+                        .setNullValue(NullValue.NULL_VALUE)
+                        .build())
+                .setDtype(DTypes.nullType())
+                .build();
+    }
+
+    static ScalarProtos.Scalar bool(boolean value) {
```

Comment:
for all of these i inherit the behavior of `Scalar::from` on the rust side

- null values are have Null ScalarValue and nullable DType
- non-null values have actual ScalarValue and non-nullable DType

We must be doing nullability coercion somewhere inside of the Rust processing